lib/Runtime/Library/JavascriptBigInt.cpp

```diff
@@ -71,21 +91,44 @@ namespace Js
         return true;
     }
 
+    bool JavascriptBigInt::Resize(digit_t length)
+    {
+        digit_t *digits;
+
+        if (length <= m_maxLength)
+            return true;
+
+        length += length;//double size
+        if (m_digits == m_reservedDigitsInit)
+        {
+            if ((SIZE_MAX / sizeof(digit_t) < length) || (NULL == (digits = (digit_t *)malloc(length * sizeof(digit_t)))))
+                return false;
+            if (0 < m_length)
+                js_memcpy_s(digits, length * sizeof(digit_t), m_digits, m_length * sizeof(digit_t));
+        }
+        else if (NULL == (digits = (digit_t *)realloc(m_digits, length * sizeof(digit_t))))
```

Comment:
Here's a thought: this business of reallocating and copying is great for all kinds of places where you don't know ahead of time how much storage you need. However, it seems to me like we always have a pretty good idea of the upper bound on the number of digits we need to build a BigInt:
- Addition, subtraction: Length of the longer input plus 1
- Multiplication: Sum of the count of digits in each input
- Parsing: Directly related to the input string length, which I think the scanner already gives us (we'd have to peek at the beginning for "0x" etc to figure out the radix).

Anyhow, not a blocker, just a thought.